ocw_oer_export/create_csv.py

```diff
@@ -0,0 +1,172 @@
+    return "|".join(topic["name"] for topic in list_of_topics_objs)
+
+
+def get_cr_authors(list_of_authors_objs):
+    """Get Course Resource Authors from a list of OCW author objects."""
+    return "|".join(
+        f"{author['last_name']}, {author['first_name']}"
+        for author in list_of_authors_objs
+    )
+
+
+def get_cr_educational_use(course_feature_tags):
+    """Get Course Resource Educational Use based on course feature tags."""
+    tags = ["Curriculum/Instruction"]
+    assessment_flag = any("Assignment" in tag for tag in course_feature_tags)
+    professional_dev_flag = "Instructor Insights" in course_feature_tags
+
+    if assessment_flag:
+        tags.append("Assessment")
+
+    if professional_dev_flag:
+        tags.append("Professional Development")
+
+    return "|".join(tags)
+
+
+def get_cr_accessibility(course_feature_tags):
+    """Get Course Resource Accessibility tags based on course feature tags."""
+    tags = ["Visual|Textual"]
+    video_flag = any("Video" in tag for tag in course_feature_tags)
+
+    if video_flag:
+        tags.append("Auditory|Caption|Transcript")
+
+    return "|".join(tags)
+
+
+def get_description_in_plain_text(description):
+    """Get Course Resource plain text description by cleaning up markdown and HTML."""
+    cleaned_description = text_cleanup(description)
+    return cleaned_description
+
+
+def transform_single_course(course, ocw_topics_mapping):
+    """Transform a single course according to OER template."""
+    if course.get("runs"):
+        return {
+            "CR_TITLE": course["title"],
+            "CR_URL": course["runs"][0]["url"],
+            "CR_MATERIAL_TYPE": "Full Course",
+            "CR_Media_Formats": "Text/HTML",
+            "CR_SUBLEVEL": "null",
+            "CR_ABSTRACT": get_description_in_plain_text(
+                course["runs"][0]["description"]
+            ),
+            "CR_LANGUAGE": "en",
+            "CR_COU_TITLE": "Creative Commons Attribution Non Commercial Share Alike 4.0",
+            "CR_PRIMARY_USER": "student|teacher",
+            "CR_SUBJECT": get_cr_subjects(ocw_topics_mapping, course["topics"]),
+            "CR_KEYWORDS": get_cr_keywords(course["topics"]),
+            "CR_AUTHOR_NAME": get_cr_authors(course["runs"][0]["instructors"]),
+            "CR_PROVIDER": "MIT",
+            "CR_PROVIDER_SET": "MIT OpenCourseWare",
+            "CR_COU_URL": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
+            "CR_COU_COPYRIGHT_HOLDER": "MIT",
+            "CR_EDUCATIONAL_USE": get_cr_educational_use(course["course_feature"]),
+            "CR_ACCESSIBILITY": get_cr_accessibility(course["course_feature"]),
+        }
+    return None
+
+
+def transform_data(data, ocw_topics_mapping):
+    """Transform all courses into OER template."""
+    return [
+        course
+        for course in (
+            transform_single_course(course, ocw_topics_mapping) for course in data
+        )
+        if course is not None
```

Comment:
Is there a reason to check for `None` here? `None` would surprise me. Maybe better to just error?

Reply:
So what happens is:

```
transform_single_course():
    if course.get("runs"):
    populate object's key-value fields and return object
    else return None
```

and then:

```
transform_data():
return [course for course if it exists]
```

if the course in API has missing `runs` data (which contains most of the needed fields), then that's problematic for us. I don't know `runs` is based off of what, but it shouldn't be empty.

`runs` was coming as empty for my test course in RC Api- `1.001-ibrahim-test`. Getting rid of this object in JSON file fixes the errors we were getting otherwise.

I don't know what particular thing this test course has missing for it to not have `runs` if you compare it to any other course:
https://github.mit.edu/ocw-content-rc/1.001-ibrahim-test
https://github.mit.edu/ocw-content-rc/24.908-spring-2017

We could just do this:

```
transform_single_course():
    if course.get("runs"):
    populate object's key-value fields and return object
```

and then this:
```
transform_data():
return [course for course]
```

But then we wouldn't know of any problematic course.
I think to decide if we should error or not, we need to find out what exactly is `runs` derived from.

Reply:
@ChristopherChudzicki 
In the production API link, we do not have any course with missing `runs` (as expected). But I'm curious, would you know why my RC test course would be missing that?